src/term.rs

```diff
@@ -71,6 +71,14 @@ impl<V: Version> Term<V> {
             _ => panic!("Negative term cannot unwrap positive range"),
         }
     }
+
+    /// Unwrap the range contains in the term.
+    pub(crate) fn as_range(&self) -> &Range<V> {
```

Comment:
I'm not a big fan of this function either. `as_range` on a term feels like it should be negated for a `Negative(range)`. But a term can never be converted into a range without loosing on the semantics. So I'd probably go for a name such as `unwrap()` or similar, meaning that some information is lost. Maybe `unwrap_range` or `extract_range`.